collect_app/src/main/java/org/odk/collect/android/utilities/EventLogger.java

```diff
@@ -1,31 +1,40 @@
 
 package org.odk.collect.android.utilities;
 
+import android.location.Location;
 import android.os.AsyncTask;
 import android.os.SystemClock;
 
 import org.javarosa.core.model.instance.TreeReference;
 import org.javarosa.form.api.FormEntryController;
-import org.odk.collect.android.logic.FormController;
-import org.odk.collect.android.tasks.TimerSaveTask;
+import org.odk.collect.android.logic.Audit;
+import org.odk.collect.android.logic.Event;
+import org.odk.collect.android.tasks.EventSaveTask;
 
 import java.io.File;
 import java.util.ArrayList;
+import java.util.List;
 
+import io.reactivex.annotations.Nullable;
 import timber.log.Timber;
 
 import static org.odk.collect.android.logic.FormController.AUDIT_FILE_NAME;
+import static org.odk.collect.android.utilities.EventLogger.EventTypes.LOCATION_PROVIDERS_DISABLED;
+import static org.odk.collect.android.utilities.EventLogger.EventTypes.LOCATION_PROVIDERS_ENABLED;
 
 /**
- * Handle logging of timer events and pass them to an Async task to append to a file
+ * Handle logging of events (which contain time and might contain location coordinates),
+ * and pass them to an Async task to append to a file
  * Notes:
  * 1) If the user has saved the form, then resumes editing, then exits without saving then the timing data during the
  * second editing session will be saved.  This is OK as it records user activity.  However if the user exits
  * without saving and they have never saved the form then the timing data is lost as the form editing will be
  * restarted from scratch.
- * 2) The times for questions in a group are not shown.  Only the time for the group is shown.
+ * 2) The events for questions in a field-list group are not shown.  Only the event for the group is shown.
  */
-public class TimerLogger {
+public class EventLogger {
+
+    private List<Location> locations = new ArrayList<>();
 
     public enum EventTypes {
         FEC,                // FEC, Real type defined in FormEntryController
```

Comment:
`FEC` is a mysterious constant name.

Reply:
It's used in one place when we create a new view. It's an event which might have different types:
EVENT_QUESTION
EVENT_GROUP
EVENT_PROMPT_NEW_REPEAT
EVENT_END_OF_FORM